modules/core/src/smithy4s/schema/Alt.scala

```diff
@@ -39,6 +41,59 @@ object Alt {
       WithValue(alt.mapK(fk), value)
   }
 
+  /**
+    * Precompiles an Alt to produce an instance of `G`
+    */
+  trait Precompiler[F[_], G[_]] { self =>
+    def apply[A](label: String, instance: F[A]): G[A]
+    def toPolyFunction[U]: PolyFunction[Alt[F, U, *], G] =
+      new PolyFunction[Alt[F, U, *], G] {
+        def apply[A](fa: Alt[F, U, A]): G[A] = self.apply(fa.label, fa.instance)
+      }
+  }
```

Comment:
**suggestion:** some helper for converting from `F ~> G` (e.g. if you want to simply lift a visitor to Precompiler) would be great (it'd just ignore the `label`)

Reply:
tbh I don't _actually_ have a need for this, so feel free to ignore